### PR TITLE
Make uniqueName require a valid identifier

### PR DESCRIPTION
**Pull request by zdevito**
With just a couple changes, we can force uniqueName to always be a valid c/python identifier. This will make it much easier to handle when serializing/de-serializing and allows uniqueName() to be used in schema.

### CODE REVIEW

File: torch/csrc/jit/ir.cpp [hunk 1]
```diff
@@ -519,11 +519,21 @@ std::shared_ptr<Graph> Graph::copy() {
...
 }
 
-Value* Value::setUniqueName(const std::string & name) {
-  if (name.size() > 0 && name.find_first_not_of("0123456789") == std::string::npos) {
-    throw std::runtime_error("names may not be integers: " + name);
+static bool isValidIdentifierChar(char c, size_t pos) {
+  return islower(c) || isupper(c) || c == '_' ||  (pos > 0 && isdigit(c));
+}
+
+static bool isValidIdentifider(const std::string & name) {
```
Comment (jamesr66a) [inline torch/csrc/jit/ir.cpp]: s/isValidIdentifider/isValidIdentifier/

**Review by jamesr66a — APPROVED** [review]
nice

**Comment by jamesr66a** [comment]
Actually, this means that if a user explicitly writes an identifier like `foo_3`, we will deduplicate this name by incrementing their index, right? So I would start seeing `foo_4`, `foo_5`, `foo_6` ?

**Comment by zdevito** [comment]
@jamesr66a Yes, which seems fine to me. When you call setUniqueName and there is already another thing with that name, the only guarantee is that the thing you are calling setUniqueName on will have that name after the call. 

I had to add more changes to make onnx still work, because it is using uniqueNames in a semantically meaningful way during export.

**Review by apaszke — APPROVED** [review]

**Comment by zdevito** [comment]
I am probably not going to merge this. It turns out that even if we enforce valid identifiers, we still end up with arbitrary other restrictions in serialization (no python keywords, for instance) and have to do further renaming there. So it seems like we don't actually need this, and would complicate the API since ONNX would still need a separate path.

**Comment by zdevito** [comment]
We no longer need our uniqueNames to be valid identifiers and this causes problems with ONNX.

**Pull request closed without merging**